src/chips/hotel/src/gpio.rs

```diff
@@ -105,15 +149,50 @@ impl hil::gpio::GPIOPin for GPIOPin {
     }
 
     fn read(&self) -> bool {
-        // TODO(alevy): implement
-        false
+        let port: &mut PortRegisters = unsafe { transmute(self.port) };
+        port.data_in.get() & (1 << (self.pin as u32)) != 0
     }
 
-    fn enable_interrupt(&self, _identifier: usize, _mode: hil::gpio::InterruptMode) {
-        // TODO(alevy): implement
+    // `InterruptMode::Change` is not implemented in hardware, so we simulate it
+    // in software. This could lead to missing events if a toggle happens before
+    // we install the new events.
```

Comment:
RE: this comment. I think this is fine. It's possible to miss interrupts with a hardware "Change" interrupt mode, just less likely since less time elapses. Not sure if there is something better to do.